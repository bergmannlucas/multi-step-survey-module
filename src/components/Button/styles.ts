import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    height: 7rem;
    width: 100%;

    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};
    text-transform: uppercase;

    color: ${theme.colors.white};
    background: ${theme.colors.purple};

    border: 4px solid ${theme.colors.purple};
    border-radius: ${theme.border.radius.huge};
    cursor: pointer;

    &:hover:enabled {
      color: ${theme.colors.purple};
      background: ${theme.colors.white};
    }

    &:disabled {
      cursor: not-allowed;
      filter: saturate(10%);
    }
  `}
`
