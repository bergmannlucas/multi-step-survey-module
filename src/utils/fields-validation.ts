import Joi from 'joi'

export const schema = Joi.array().items(
  Joi.object({
    name: Joi.string().empty(''),
    email: Joi.string()
      .email({ tlds: { allow: false } })
      .empty('')
  }),
  Joi.object({
    gender: Joi.string().required(),
    age: Joi.number().required()
  }),
  Joi.object({
    book: Joi.string().required(),
    colors: Joi.object({
      black: Joi.boolean(),
      white: Joi.boolean(),
      red: Joi.boolean(),
      green: Joi.boolean(),
      blue: Joi.boolean(),
      yellow: Joi.boolean(),
      purple: Joi.boolean(),
      orange: Joi.boolean(),
      pink: Joi.boolean()
    })
  })
)
