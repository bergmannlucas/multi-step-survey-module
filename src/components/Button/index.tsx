import { ButtonHTMLAttributes } from 'react'

import * as S from './styles'

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement>

function Button({ children, ...props }: ButtonProps) {
  return (
    <S.Wrapper {...props}>
      <span>{children}</span>
    </S.Wrapper>
  )
}

export default Button
