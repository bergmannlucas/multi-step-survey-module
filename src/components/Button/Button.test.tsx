import { render, screen } from 'utils/test-utils'
import Button from '.'

describe('<Button />', () => {
  it('should render correctly', () => {
    const { container } = render(<Button>Next</Button>)

    expect(screen.getByText(/next/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a disabled Button', () => {
    render(<Button disabled>Submit</Button>)

    expect(screen.getByRole('button', { name: /submit/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled'
      }
    )
  })
})
