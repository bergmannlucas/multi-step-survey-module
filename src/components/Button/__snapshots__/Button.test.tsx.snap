// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.c0 {
  height: 7rem;
  width: 100%;
  font-family: Montserrat,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  text-transform: uppercase;
  color: #FAFAFA;
  background: #9249FF;
  border: 4px solid #9249FF;
  border-radius: 3.5rem;
  cursor: pointer;
}

.c0:hover:enabled {
  color: #9249FF;
  background: #FAFAFA;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: saturate(10%);
  filter: saturate(10%);
}

<button
  class="c0"
>
  <span>
    Next
  </span>
</button>
`;
